src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -82,31 +85,36 @@ def isEnabled(self, service):
         return enabled
 
     @private
-    def tryConnect(self, host, port):
+    def tryConnect(self, host, port, fnldap):
         max_tries = 3
         connected = False
 
-        timeout = _fs().middlewared.plugins.service_monitor.socket_timeout
+        if self.name == 'activedirectory':
+            host_list = []
 
-        for i in range(0, max_tries):
-            # XXX What about UDP?
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            s.settimeout(timeout)
-
-            try:
-                s.connect((host, port))
-                connected = True
+            for i in range(0, max_tries):
+                # Make max_tries attempts to get SRV records from DNS
+                host_list = fnldap.get_ldap_servers(host)
+                if host_list:
+                    break
+                else:
+                    self.logger.debug("[ServiceMonitorThread] Attempt %d to query SRV records failed " % (i))
```

Comment:
Please, try to avoid legacy `"%x" % (v)` syntax. Use either `f'{v}'` or `'{}'.format(v)` instead.